### PR TITLE
feat: Custom application icon

### PR DESCRIPTION
**Pull request by carlosjgp**
Relates to #2961 

I'm trying to add a field to the UI forms and components to allow to render a custom icon for apps and default to Helm `Chart.yaml` icon field if it's defined but NodeJs is not my strength
and I'm struggling to figure out how to get/post the information from the two different components (`application` and `server`) and where on the UI this information should be merged

I'll post a message on the Slack channel and use this PR to share the code

Signed-off-by: Carlos Juan Gómez Peñalver <carlosjuangp@gmail.com>

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

**Review by jannfis — CHANGES_REQUESTED** [review]
Hi @carlosjgp, thanks for submitting your PR, it's much appreciated!

While I think the change you are introducing indeed is useful, I feel a little uncomfortable with having the UI display images from possibly untrusted sources. I think that this feature should be toggleable, maybe on a project level, to prevent unwanted tracking of UI users. Maybe even go so far to whitelist certain URLs usable as icon source. I'm not sure whether I'm being too paranoid here, so I'd love to hear other opinions @alexmt @jessesuen 

Also, please find some minor things I found during my review in the comments below .

File: ui/.node-version [hunk 1]
```diff
@@ -0,0 +1 @@
+v11.15.0
```
[the remark this answers is not in the document]
Comment (carlosjgp) [reply ui/.node-version]: I use AVN to switch between NodeJS version on different repository automatically
https://github.com/wbyoung/avn

Advised by my company's NodeJS developers (I'm on the infrastructure/DevOps team)
I didn't mean to commit this file

I'll remove it!

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -30,9 +30,9 @@ brew install go git-lfs kubectl kubectx dep ksonnet/tap/ks helm@2 kustomize
 Check the versions:
 
 ```bash
-go version ;# must be v1.12.x
-helm version ;# must be v2.13.x
-kustomize version ;# must be v3.1.x
+go version ;# must be v1.14.x
+helm version ;# must be v3.1.x
+kustomize version ;# must be v3.2.x
 ```
```
[the remark this answers is not in the document]
Comment (carlosjgp) [reply docs/CONTRIBUTING.md]: I'll raise a separate PR to update the documentation and add the instructions for `helm2`

File: pkg/apiclient/repository/repository.pb.go [hunk 3]
```diff
@@ -587,7 +587,9 @@ func init() {
 	proto.RegisterType((*RepoUpdateRequest)(nil), "repository.RepoUpdateRequest")
 }
 
-func init() { proto.RegisterFile("server/repository/repository.proto", fileDescriptor_8d38260443475705) }
+func init() {
+	proto.RegisterFile("server/repository/repository.proto", fileDescriptor_8d38260443475705)
+}
```
[the remark this answers is not in the document]
Comment (carlosjgp) [reply pkg/apiclient/repository/repository.pb.go]: Yeah, I tried to use it to update the NodeJS models but it seems to be a manual process...

I tried
```
make protogen codegen codegen-local                                                                                                                                                                
make codegen openapigen                                                                                                                                                                            
make codegen lint test cli image 
```

I'll revert that

**Comment by carlosjgp** [comment]
Looks like the contributing guidelines have been moved it looks very nice!

and this PR is now a little bit behind `master`. I'll close it and will start over again and maybe raise a draft PR which I just discovered while starting to contribute to [Terraform](https://github.com/hashicorp/terraform/blob/master/.github/CONTRIBUTING.md#pull-request-lifecycle)
https://github.blog/2019-02-14-introducing-draft-pull-requests/

Something that might interest Argo Project Team to add to the contributing guidelines?

Thanks for your time and comments @jannfis

**Pull request closed without merging**